# Different operation in sql
[lecture link](https://www.youtube.com/watch?v=9mjLMw9nwJI)

**Date 13 nov 2022**

Update,delete, filter and many others operation.

sql is very powerful language.   
In India and Europe sql is very important.

In [1]:
import mysql.connector as conn

In [2]:
mydb = conn.connect(host = 'localhost', user = 'root', passwd = '123456')

In [3]:
cursor = mydb.cursor()

In [4]:
cursor.execute('show databases')
cursor.fetchall()

[('employees',),
 ('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sales',),
 ('sharad',),
 ('sharad1',)]

In [5]:
cursor.execute('select * from sharad.glassdata')
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (2, 1.51761, 13.89, 3.6, 1.36, 72.73, 0.48, 7.83, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (5, 1.51742, 13.27, 3.62, 1.24, 73.08, 0.55, 8.07, 0.0, 0.0, 1),
 (6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1),
 (7, 1.51743, 13.3, 3.6, 1.14, 73.09, 0.58, 8.17, 0.0, 0.0, 1),
 (8, 1.51756, 13.15, 3.61, 1.05, 73.24, 0.57, 8.24, 0.0, 0.0, 1),
 (9, 1.51918, 14.04, 3.58, 1.37, 72.08, 0.56, 8.3, 0.0, 0.0, 1),
 (10, 1.51755, 13.0, 3.6, 1.36, 72.99, 0.57, 8.4, 0.0, 0.11, 1),
 (11, 1.51571, 12.72, 3.46, 1.56, 73.2, 0.67, 8.09, 0.0, 0.24, 1),
 (12, 1.51763, 12.8, 3.66, 1.27, 73.01, 0.6, 8.56, 0.0, 0.0, 1),
 (13, 1.51589, 12.88, 3.43, 1.4, 73.28, 0.69, 8.05, 0.0, 0.24, 1),
 (14, 1.51748, 12.86, 3.56, 1.27, 73.21, 0.54, 8.38, 0.0, 0.17, 1),
 (15, 1.51763, 12.61, 3.59, 1.31, 73.29, 0.58, 8.5, 0.0, 0.0, 1),
 (16, 1.51761, 12.81, 3.54,

In [6]:
cursor.execute('select * from sharad.ineuron')
cursor.fetchall()

[(20071, 'Sharad', 'Pardhe', 20071, 'FSDS'),
 (20071, 'Sharad', 'Pardhe', 20071, 'FSDS')]

In [7]:
cursor.execute('select col1,col1, col5, col10,col1 from sharad.glassdata')
cursor.fetchall()

[(6, 6, 1.1, 0.0, 6),
 (6, 6, 1.1, 0.0, 6),
 (2, 2, 1.36, 0.0, 2),
 (3, 3, 1.54, 0.0, 3),
 (5, 5, 1.24, 0.0, 5),
 (6, 6, 1.62, 0.26, 6),
 (7, 7, 1.14, 0.0, 7),
 (8, 8, 1.05, 0.0, 8),
 (9, 9, 1.37, 0.0, 9),
 (10, 10, 1.36, 0.11, 10),
 (11, 11, 1.56, 0.24, 11),
 (12, 12, 1.27, 0.0, 12),
 (13, 13, 1.4, 0.24, 13),
 (14, 14, 1.27, 0.17, 14),
 (15, 15, 1.31, 0.0, 15),
 (16, 16, 1.23, 0.0, 16),
 (17, 17, 1.16, 0.0, 17),
 (18, 18, 0.89, 0.0, 18),
 (19, 19, 1.18, 0.0, 19),
 (20, 20, 1.69, 0.07, 20),
 (21, 21, 1.49, 0.19, 21),
 (22, 22, 0.29, 0.0, 22),
 (23, 23, 1.29, 0.0, 23),
 (24, 24, 1.35, 0.0, 24),
 (25, 25, 1.15, 0.0, 25),
 (26, 26, 1.21, 0.0, 26),
 (27, 27, 1.41, 0.0, 27),
 (28, 28, 1.33, 0.0, 28),
 (29, 29, 1.43, 0.0, 29),
 (30, 30, 1.28, 0.0, 30),
 (31, 31, 1.3, 0.14, 31),
 (32, 32, 1.14, 0.0, 32),
 (33, 33, 1.23, 0.22, 33),
 (34, 34, 1.38, 0.06, 34),
 (35, 35, 1.34, 0.0, 35),
 (36, 36, 1.21, 0.0, 36),
 (37, 37, 1.32, 0.0, 37),
 (38, 38, 1.35, 0.0, 38),
 (39, 39, 0.47, 0.0, 39),
 (40, 4

In [8]:
cursor.execute("select * from sharad.glassdata where col1 = 4")
cursor.fetchall()

[]

In [9]:
cursor.execute("select * from sharad.glassdata where col1 = 4 and col11 = 0")
cursor.fetchall()

[]

NO such values exists.

**Que 1** Delete the data where col1 = 4?

In [10]:
cursor.execute("delete from sharad.glassdata where col1 = 4")
mydb.commit()

In [11]:
cursor.execute('select* from sharad.glassdata')
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (2, 1.51761, 13.89, 3.6, 1.36, 72.73, 0.48, 7.83, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (5, 1.51742, 13.27, 3.62, 1.24, 73.08, 0.55, 8.07, 0.0, 0.0, 1),
 (6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1),
 (7, 1.51743, 13.3, 3.6, 1.14, 73.09, 0.58, 8.17, 0.0, 0.0, 1),
 (8, 1.51756, 13.15, 3.61, 1.05, 73.24, 0.57, 8.24, 0.0, 0.0, 1),
 (9, 1.51918, 14.04, 3.58, 1.37, 72.08, 0.56, 8.3, 0.0, 0.0, 1),
 (10, 1.51755, 13.0, 3.6, 1.36, 72.99, 0.57, 8.4, 0.0, 0.11, 1),
 (11, 1.51571, 12.72, 3.46, 1.56, 73.2, 0.67, 8.09, 0.0, 0.24, 1),
 (12, 1.51763, 12.8, 3.66, 1.27, 73.01, 0.6, 8.56, 0.0, 0.0, 1),
 (13, 1.51589, 12.88, 3.43, 1.4, 73.28, 0.69, 8.05, 0.0, 0.24, 1),
 (14, 1.51748, 12.86, 3.56, 1.27, 73.21, 0.54, 8.38, 0.0, 0.17, 1),
 (15, 1.51763, 12.61, 3.59, 1.31, 73.29, 0.58, 8.5, 0.0, 0.0, 1),
 (16, 1.51761, 12.81, 3.54,

col1 = 4 column has been deleted

**Update operation** 

In [12]:
cursor.execute('UPDATE sharad.glassdata SET col2 = 6, col3 = 8 WHERE col1 = 3')
mydb.commit()

In [13]:
cursor.execute('select * from sharad.glassdata WHERE col1 = 3')
cursor.fetchall()

[(3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1)]

mission success 

In [14]:
cursor.execute('select * from sharad.glassdata where col1 = 6')
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1)]

**Group by operation**

**15 Nov 2022**

In [15]:
quary = 'select count(col1),col2 from sharad.glassdata group by col1'
cursor.execute(quary)
cursor.fetchall()

[(7, 8.0),
 (3, 1.51761),
 (3, 6.0),
 (3, 1.51742),
 (3, 1.51743),
 (3, 1.51756),
 (3, 1.51918),
 (3, 1.51755),
 (3, 1.51571),
 (3, 1.51763),
 (3, 1.51589),
 (3, 1.51748),
 (3, 1.51763),
 (3, 1.51761),
 (3, 1.51784),
 (3, 1.52196),
 (3, 1.51911),
 (3, 1.51735),
 (3, 1.5175),
 (3, 1.51966),
 (3, 1.51736),
 (3, 1.51751),
 (3, 1.5172),
 (3, 1.51764),
 (3, 1.51793),
 (3, 1.51721),
 (3, 1.51768),
 (3, 1.51784),
 (3, 1.51768),
 (3, 1.51747),
 (3, 1.51775),
 (3, 1.51753),
 (3, 1.51783),
 (3, 1.51567),
 (3, 1.51909),
 (3, 1.51797),
 (3, 1.52213),
 (3, 1.52213),
 (3, 1.51793),
 (3, 1.51755),
 (3, 1.51779),
 (3, 1.5221),
 (3, 1.51786),
 (3, 1.519),
 (3, 1.51869),
 (3, 1.52667),
 (3, 1.52223),
 (3, 1.51898),
 (3, 1.5232),
 (3, 1.51926),
 (3, 1.51808),
 (3, 1.51837),
 (3, 1.51778),
 (3, 1.51769),
 (3, 1.51215),
 (3, 1.51824),
 (3, 1.51754),
 (3, 1.51754),
 (3, 1.51905),
 (3, 1.51977),
 (3, 1.52172),
 (3, 1.52227),
 (3, 1.52172),
 (3, 1.52099),
 (3, 1.52152),
 (3, 1.52152),
 (3, 1.52152),
 (3, 1.52

In [16]:
quary = 'select col1,count(col1) from sharad.glassdata group by col1'
cursor.execute(quary)
cursor.fetchall()

[(6, 7),
 (2, 3),
 (3, 3),
 (5, 3),
 (7, 3),
 (8, 3),
 (9, 3),
 (10, 3),
 (11, 3),
 (12, 3),
 (13, 3),
 (14, 3),
 (15, 3),
 (16, 3),
 (17, 3),
 (18, 3),
 (19, 3),
 (20, 3),
 (21, 3),
 (22, 3),
 (23, 3),
 (24, 3),
 (25, 3),
 (26, 3),
 (27, 3),
 (28, 3),
 (29, 3),
 (30, 3),
 (31, 3),
 (32, 3),
 (33, 3),
 (34, 3),
 (35, 3),
 (36, 3),
 (37, 3),
 (38, 3),
 (39, 3),
 (40, 3),
 (41, 3),
 (42, 3),
 (43, 3),
 (44, 3),
 (45, 3),
 (46, 3),
 (47, 3),
 (48, 3),
 (49, 3),
 (50, 3),
 (51, 3),
 (52, 3),
 (53, 3),
 (54, 3),
 (55, 3),
 (56, 3),
 (57, 3),
 (58, 3),
 (59, 3),
 (60, 3),
 (61, 3),
 (62, 3),
 (63, 3),
 (64, 3),
 (65, 3),
 (66, 3),
 (67, 3),
 (68, 3),
 (69, 3),
 (70, 3),
 (71, 3),
 (72, 3),
 (73, 3),
 (74, 3),
 (75, 3),
 (76, 3),
 (77, 3),
 (78, 3),
 (79, 3),
 (80, 3),
 (81, 3),
 (82, 3),
 (83, 3),
 (84, 3),
 (85, 3),
 (86, 3),
 (87, 3),
 (88, 3),
 (89, 3),
 (90, 3),
 (91, 3),
 (92, 3),
 (93, 3),
 (94, 3),
 (95, 3),
 (96, 3),
 (97, 3),
 (98, 3),
 (99, 3),
 (100, 3),
 (101, 3),
 (102, 3),
 (10

In [17]:
quary = 'select * from sharad.glassdata'
cursor.execute(quary)
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (2, 1.51761, 13.89, 3.6, 1.36, 72.73, 0.48, 7.83, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (5, 1.51742, 13.27, 3.62, 1.24, 73.08, 0.55, 8.07, 0.0, 0.0, 1),
 (6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1),
 (7, 1.51743, 13.3, 3.6, 1.14, 73.09, 0.58, 8.17, 0.0, 0.0, 1),
 (8, 1.51756, 13.15, 3.61, 1.05, 73.24, 0.57, 8.24, 0.0, 0.0, 1),
 (9, 1.51918, 14.04, 3.58, 1.37, 72.08, 0.56, 8.3, 0.0, 0.0, 1),
 (10, 1.51755, 13.0, 3.6, 1.36, 72.99, 0.57, 8.4, 0.0, 0.11, 1),
 (11, 1.51571, 12.72, 3.46, 1.56, 73.2, 0.67, 8.09, 0.0, 0.24, 1),
 (12, 1.51763, 12.8, 3.66, 1.27, 73.01, 0.6, 8.56, 0.0, 0.0, 1),
 (13, 1.51589, 12.88, 3.43, 1.4, 73.28, 0.69, 8.05, 0.0, 0.24, 1),
 (14, 1.51748, 12.86, 3.56, 1.27, 73.21, 0.54, 8.38, 0.0, 0.17, 1),
 (15, 1.51763, 12.61, 3.59, 1.31, 73.29, 0.58, 8.5, 0.0, 0.0, 1),
 (16, 1.51761, 12.81, 3.54,

In [18]:
quary = 'select col1,count(col1) from sharad.glassdata group by col1 order by col1'
cursor.execute(quary)
cursor.fetchall()

[(2, 3),
 (3, 3),
 (5, 3),
 (6, 7),
 (7, 3),
 (8, 3),
 (9, 3),
 (10, 3),
 (11, 3),
 (12, 3),
 (13, 3),
 (14, 3),
 (15, 3),
 (16, 3),
 (17, 3),
 (18, 3),
 (19, 3),
 (20, 3),
 (21, 3),
 (22, 3),
 (23, 3),
 (24, 3),
 (25, 3),
 (26, 3),
 (27, 3),
 (28, 3),
 (29, 3),
 (30, 3),
 (31, 3),
 (32, 3),
 (33, 3),
 (34, 3),
 (35, 3),
 (36, 3),
 (37, 3),
 (38, 3),
 (39, 3),
 (40, 3),
 (41, 3),
 (42, 3),
 (43, 3),
 (44, 3),
 (45, 3),
 (46, 3),
 (47, 3),
 (48, 3),
 (49, 3),
 (50, 3),
 (51, 3),
 (52, 3),
 (53, 3),
 (54, 3),
 (55, 3),
 (56, 3),
 (57, 3),
 (58, 3),
 (59, 3),
 (60, 3),
 (61, 3),
 (62, 3),
 (63, 3),
 (64, 3),
 (65, 3),
 (66, 3),
 (67, 3),
 (68, 3),
 (69, 3),
 (70, 3),
 (71, 3),
 (72, 3),
 (73, 3),
 (74, 3),
 (75, 3),
 (76, 3),
 (77, 3),
 (78, 3),
 (79, 3),
 (80, 3),
 (81, 3),
 (82, 3),
 (83, 3),
 (84, 3),
 (85, 3),
 (86, 3),
 (87, 3),
 (88, 3),
 (89, 3),
 (90, 3),
 (91, 3),
 (92, 3),
 (93, 3),
 (94, 3),
 (95, 3),
 (96, 3),
 (97, 3),
 (98, 3),
 (99, 3),
 (100, 3),
 (101, 3),
 (102, 3),
 (10

In [19]:
q = 'select * from sharad.ineuron'
cursor.execute(q)
cursor.fetchall()

[(20071, 'Sharad', 'Pardhe', 20071, 'FSDS'),
 (20071, 'Sharad', 'Pardhe', 20071, 'FSDS')]

In [20]:
q = 'select class_name,count(class_name) from sharad.ineuron group by class_name'
cursor.execute(q)
cursor.fetchall()

[('FSDS', 2)]

In [21]:
q = 'SELECT first_name,COUNT(first_name) FROM employees.employees Group by first_name'
cursor.execute(q)
cursor.fetchall()

[('Georgi', 253),
 ('Bezalel', 228),
 ('Parto', 228),
 ('Chirstian', 226),
 ('Kyoichi', 251),
 ('Anneke', 225),
 ('Tzvetan', 241),
 ('Saniya', 257),
 ('Sumant', 249),
 ('Duangkaew', 226),
 ('Mary', 224),
 ('Patricio', 237),
 ('Eberhardt', 246),
 ('Berni', 234),
 ('Guoxiang', 255),
 ('Kazuhito', 245),
 ('Cristinel', 221),
 ('Kazuhide', 234),
 ('Lillian', 243),
 ('Mayuko', 259),
 ('Ramzi', 265),
 ('Shahaf', 229),
 ('Bojan', 263),
 ('Suzette', 242),
 ('Prasadram', 242),
 ('Yongqiao', 249),
 ('Divier', 247),
 ('Domenick', 228),
 ('Otmar', 232),
 ('Elvis', 246),
 ('Karsten', 242),
 ('Jeong', 239),
 ('Arif', 204),
 ('Bader', 202),
 ('Alain', 240),
 ('Adamantios', 206),
 ('Pradeep', 227),
 ('Huan', 242),
 ('Alejandro', 211),
 ('Weiyi', 224),
 ('Uri', 225),
 ('Magy', 223),
 ('Yishay', 217),
 ('Mingsen', 249),
 ('Moss', 245),
 ('Lucien', 227),
 ('Zvonko', 258),
 ('Florian', 222),
 ('Basil', 236),
 ('Yinghua', 242),
 ('Hidefumi', 222),
 ('Heping', 239),
 ('Sanjiv', 212),
 ('Mayumi', 253),
 ('Geo

In [22]:
q = 'drop table sharad.ineuron'
cursor.execute(q)
mydb.commit()

In [23]:
q = 'select * from sharad.ineuron'
cursor.execute(q)
cursor.fetchall()

ProgrammingError: 1146 (42S02): Table 'sharad.ineuron' doesn't exist

**ProgrammingError:** 1146 (42S02): Table 'sharad.ineuron' doesn't exist.    
```sharad.ineuron``` has been deleted 

In [26]:
q = 'select * from sharad.glassdata'
cursor.execute(q)
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (2, 1.51761, 13.89, 3.6, 1.36, 72.73, 0.48, 7.83, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (5, 1.51742, 13.27, 3.62, 1.24, 73.08, 0.55, 8.07, 0.0, 0.0, 1),
 (6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1),
 (7, 1.51743, 13.3, 3.6, 1.14, 73.09, 0.58, 8.17, 0.0, 0.0, 1),
 (8, 1.51756, 13.15, 3.61, 1.05, 73.24, 0.57, 8.24, 0.0, 0.0, 1),
 (9, 1.51918, 14.04, 3.58, 1.37, 72.08, 0.56, 8.3, 0.0, 0.0, 1),
 (10, 1.51755, 13.0, 3.6, 1.36, 72.99, 0.57, 8.4, 0.0, 0.11, 1),
 (11, 1.51571, 12.72, 3.46, 1.56, 73.2, 0.67, 8.09, 0.0, 0.24, 1),
 (12, 1.51763, 12.8, 3.66, 1.27, 73.01, 0.6, 8.56, 0.0, 0.0, 1),
 (13, 1.51589, 12.88, 3.43, 1.4, 73.28, 0.69, 8.05, 0.0, 0.24, 1),
 (14, 1.51748, 12.86, 3.56, 1.27, 73.21, 0.54, 8.38, 0.0, 0.17, 1),
 (15, 1.51763, 12.61, 3.59, 1.31, 73.29, 0.58, 8.5, 0.0, 0.0, 1),
 (16, 1.51761, 12.81, 3.54,

In [27]:
q = 'select col1,col2,col11 from sharad.glassdata where col1=11 or col11 =2'
cursor.execute(q)
cursor.fetchall()

[(11, 1.51571, 1),
 (71, 1.51574, 2),
 (72, 1.51848, 2),
 (73, 1.51593, 2),
 (74, 1.51631, 2),
 (75, 1.51596, 2),
 (76, 1.5159, 2),
 (77, 1.51645, 2),
 (78, 1.51627, 2),
 (79, 1.51613, 2),
 (80, 1.5159, 2),
 (81, 1.51592, 2),
 (82, 1.51593, 2),
 (83, 1.51646, 2),
 (84, 1.51594, 2),
 (85, 1.51409, 2),
 (86, 1.51625, 2),
 (87, 1.51569, 2),
 (88, 1.51645, 2),
 (89, 1.51618, 2),
 (90, 1.5164, 2),
 (91, 1.51841, 2),
 (92, 1.51605, 2),
 (93, 1.51588, 2),
 (94, 1.5159, 2),
 (95, 1.51629, 2),
 (96, 1.5186, 2),
 (97, 1.51841, 2),
 (98, 1.51743, 2),
 (99, 1.51689, 2),
 (100, 1.51811, 2),
 (101, 1.51655, 2),
 (102, 1.5173, 2),
 (103, 1.5182, 2),
 (104, 1.52725, 2),
 (105, 1.5241, 2),
 (106, 1.52475, 2),
 (107, 1.53125, 2),
 (108, 1.53393, 2),
 (109, 1.52222, 2),
 (110, 1.51818, 2),
 (111, 1.52664, 2),
 (112, 1.52739, 2),
 (113, 1.52777, 2),
 (114, 1.51892, 2),
 (115, 1.51847, 2),
 (116, 1.51846, 2),
 (117, 1.51829, 2),
 (118, 1.51708, 2),
 (119, 1.51673, 2),
 (120, 1.51652, 2),
 (121, 1.51844, 2)

In [31]:
q = 'select col1,col2,col11 from sharad.glassdata where col1=11 and col11 =2'
cursor.execute(q)
cursor.fetchall()

[]

No such column 

In [35]:
q = "select * from sharad.glassdata where col2 like '1.515%'"
cursor.execute(q)
cursor.fetchall()

[(6, 1.51596, 12.79, 3.61, 1.62, 72.97, 0.64, 8.07, 0.0, 0.26, 1),
 (11, 1.51571, 12.72, 3.46, 1.56, 73.2, 0.67, 8.09, 0.0, 0.24, 1),
 (13, 1.51589, 12.88, 3.43, 1.4, 73.28, 0.69, 8.05, 0.0, 0.24, 1),
 (36, 1.51567, 13.29, 3.45, 1.21, 72.74, 0.56, 8.57, 0.0, 0.0, 1),
 (71, 1.51574, 14.86, 3.67, 1.74, 71.87, 0.16, 7.36, 0.0, 0.12, 2),
 (73, 1.51593, 13.09, 3.59, 1.52, 73.1, 0.67, 7.83, 0.0, 0.0, 2),
 (75, 1.51596, 13.02, 3.56, 1.54, 73.11, 0.72, 7.9, 0.0, 0.0, 2),
 (76, 1.5159, 13.02, 3.58, 1.51, 73.12, 0.69, 7.96, 0.0, 0.0, 2),
 (80, 1.5159, 12.82, 3.52, 1.9, 72.86, 0.69, 7.97, 0.0, 0.0, 2),
 (81, 1.51592, 12.86, 3.52, 2.12, 72.66, 0.69, 7.97, 0.0, 0.0, 2),
 (82, 1.51593, 13.25, 3.45, 1.43, 73.17, 0.61, 7.86, 0.0, 0.0, 2),
 (84, 1.51594, 13.09, 3.52, 1.55, 72.87, 0.68, 8.05, 0.0, 0.09, 2),
 (87, 1.51569, 13.24, 3.49, 1.47, 73.25, 0.38, 8.03, 0.0, 0.0, 2),
 (93, 1.51588, 13.12, 3.41, 1.58, 73.26, 0.07, 8.39, 0.0, 0.19, 2),
 (94, 1.5159, 13.24, 3.34, 1.47, 73.1, 0.39, 8.22, 0.0, 0.0, 2),

In [37]:
q = 'select * from sharad.glassdata order by col2 DESC'
cursor.execute(q)
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (108, 1.53393, 12.3, 0.0, 1.0, 70.16, 0.12, 16.19, 0.0, 0.24, 2),
 (108, 1.53393, 12.3, 0.0, 1.0, 70.16, 0.12, 16.19, 0.0, 0.24, 2),
 (108, 1.53393, 12.3, 0.0, 1.0, 70.16, 0.12, 16.19, 0.0, 0.24, 2),
 (107, 1.53125, 10.73, 0.0, 2.1, 69.81, 0.58, 13.3, 3.15, 0.28, 2),
 (107, 1.53125, 10.73, 0.0, 2.1, 69.81, 0.58, 13.3, 3.15, 0.28, 2),
 (107, 1.53125, 10.73, 0.0, 2.1, 69.81, 0.58, 13.3, 3.15, 0.28, 2),
 (113, 1.52777, 12.64, 0.0, 0.67, 72.02, 0.06, 14.4, 0.0, 0.0, 2),
 (113, 1.52777, 12.64, 0.0, 0.67, 72.02, 0.06, 14.4, 0.0, 0.0, 2),
 (113, 1.52777, 12.64, 0.0, 0.67, 72.

In [39]:
q = 'select * from sharad.glassdata group by col1 '
cursor.execute(q)
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1),
 (2, 1.51761, 13.89, 3.6, 1.36, 72.73, 0.48, 7.83, 0.0, 0.0, 1),
 (3, 6.0, 8.0, 3.55, 1.54, 72.99, 0.39, 7.78, 0.0, 0.0, 1),
 (5, 1.51742, 13.27, 3.62, 1.24, 73.08, 0.55, 8.07, 0.0, 0.0, 1),
 (7, 1.51743, 13.3, 3.6, 1.14, 73.09, 0.58, 8.17, 0.0, 0.0, 1),
 (8, 1.51756, 13.15, 3.61, 1.05, 73.24, 0.57, 8.24, 0.0, 0.0, 1),
 (9, 1.51918, 14.04, 3.58, 1.37, 72.08, 0.56, 8.3, 0.0, 0.0, 1),
 (10, 1.51755, 13.0, 3.6, 1.36, 72.99, 0.57, 8.4, 0.0, 0.11, 1),
 (11, 1.51571, 12.72, 3.46, 1.56, 73.2, 0.67, 8.09, 0.0, 0.24, 1),
 (12, 1.51763, 12.8, 3.66, 1.27, 73.01, 0.6, 8.56, 0.0, 0.0, 1),
 (13, 1.51589, 12.88, 3.43, 1.4, 73.28, 0.69, 8.05, 0.0, 0.24, 1),
 (14, 1.51748, 12.86, 3.56, 1.27, 73.21, 0.54, 8.38, 0.0, 0.17, 1),
 (15, 1.51763, 12.61, 3.59, 1.31, 73.29, 0.58, 8.5, 0.0, 0.0, 1),
 (16, 1.51761, 12.81, 3.54, 1.23, 73.24, 0.58, 8.39, 0.0, 0.0, 1),
 (17, 1.51784, 12.68, 3.67, 1.16, 73.11, 0.61, 8.7, 0.0, 0.0, 1),
 (18, 1.52196, 14.36, 

In [42]:
q = 'select * from sharad.glassdata group by col1 having count(col1)>4'
cursor.execute(q)
cursor.fetchall()

[(6, 8.0, 13.64, 4.49, 1.1, 71.78, 0.06, 8.75, 0.0, 0.0, 1)]

# Questions
1. Take dataset from attached [URL](https://archive.ics.uci.edu/ml/machine-learning-databases/car/car.data)?
1. Create dataset cardata?
1. create table as car with column name given in [description](https://archive.ics.uci.edu/ml/machine-learning-databases/car/car.names)?
1. Try to check weather all the date available inside table or not?
1. Try to group all the data with col1 and count occurance of each and every record based on col1 value?
1. Try to filter where col3 value will be 4?
1. Try to update a record where col3 value with 8 whereever you have value equal to 2?
1. Try to dalete table?
1. Try to delete database?

In [4]:
ls

 Volume in drive C is Windows SSD
 Volume Serial Number is FCA4-BF44

 Directory of C:\Users\HP\1AI iNeuron

17-11-2022  09:58 PM    <DIR>          .
17-11-2022  09:58 PM    <DIR>          ..
17-11-2022  03:05 PM    <DIR>          .ipynb_checkpoints
07-11-2022  08:11 AM    <DIR>          __pycache__
15-11-2022  04:53 PM           254,536 017_sql_and_sqlConnector.ipynb
17-11-2022  09:58 PM           473,882 018_SQL_PART2_13 nov 2022.ipynb
16-11-2022  08:47 AM           465,385 018_SQL_PART2_13 nov 2022-Copy1.ipynb
17-11-2022  09:41 PM            99,986 019_Introduction_to_MongoDB_17 Nov 2022.ipynb
16-11-2022  03:34 PM               589 019_OOPS-DISCUSSION_16NOV.ipynb
17-11-2022  06:21 PM    <DIR>          2. Working with Databases & Python
17-11-2022  10:11 AM            25,496 Aeroplane.png
12-11-2022  09:32 AM             3,578 Assignment_010.ipynb
15-11-2022  10:08 AM             8,317 Assignment_011.ipynb
16-11-2022  09:55 AM             5,597 Assignment_012.ipynb
17-11-2022  09:17 

In [5]:
import mysql.connector as conn
mydb = conn.connect(host='localhost', user = 'root', passwd = '123456')
curr = mydb.cursor()

In [6]:
curr.execute('create database if not exists cardata')
curr.execute('create table cardata.car (buying varchar(255) , maint text , doors text , persons text , lug_boot text , safety text , class text )')
mydb.commit()
curr.execute('show databases')
curr.fetchall()

[('cardata',),
 ('employees',),
 ('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sales',),
 ('sharad',)]

In [7]:
curr.execute('select * from cardata.car')
curr.fetchall()

[]

In [5]:
import csv
with open("car.data",'r') as f:
    cardata = csv.reader(f, delimiter = '\n')
    for i in cardata:
        print(i)

['vhigh,vhigh,2,2,small,low,unacc']
['vhigh,vhigh,2,2,small,med,unacc']
['vhigh,vhigh,2,2,small,high,unacc']
['vhigh,vhigh,2,2,med,low,unacc']
['vhigh,vhigh,2,2,med,med,unacc']
['vhigh,vhigh,2,2,med,high,unacc']
['vhigh,vhigh,2,2,big,low,unacc']
['vhigh,vhigh,2,2,big,med,unacc']
['vhigh,vhigh,2,2,big,high,unacc']
['vhigh,vhigh,2,4,small,low,unacc']
['vhigh,vhigh,2,4,small,med,unacc']
['vhigh,vhigh,2,4,small,high,unacc']
['vhigh,vhigh,2,4,med,low,unacc']
['vhigh,vhigh,2,4,med,med,unacc']
['vhigh,vhigh,2,4,med,high,unacc']
['vhigh,vhigh,2,4,big,low,unacc']
['vhigh,vhigh,2,4,big,med,unacc']
['vhigh,vhigh,2,4,big,high,unacc']
['vhigh,vhigh,2,more,small,low,unacc']
['vhigh,vhigh,2,more,small,med,unacc']
['vhigh,vhigh,2,more,small,high,unacc']
['vhigh,vhigh,2,more,med,low,unacc']
['vhigh,vhigh,2,more,med,med,unacc']
['vhigh,vhigh,2,more,med,high,unacc']
['vhigh,vhigh,2,more,big,low,unacc']
['vhigh,vhigh,2,more,big,med,unacc']
['vhigh,vhigh,2,more,big,high,unacc']
['vhigh,vhigh,3,2,small,low,

['low,high,3,more,med,med,acc']
['low,high,3,more,med,high,vgood']
['low,high,3,more,big,low,unacc']
['low,high,3,more,big,med,acc']
['low,high,3,more,big,high,vgood']
['low,high,4,2,small,low,unacc']
['low,high,4,2,small,med,unacc']
['low,high,4,2,small,high,unacc']
['low,high,4,2,med,low,unacc']
['low,high,4,2,med,med,unacc']
['low,high,4,2,med,high,unacc']
['low,high,4,2,big,low,unacc']
['low,high,4,2,big,med,unacc']
['low,high,4,2,big,high,unacc']
['low,high,4,4,small,low,unacc']
['low,high,4,4,small,med,acc']
['low,high,4,4,small,high,acc']
['low,high,4,4,med,low,unacc']
['low,high,4,4,med,med,acc']
['low,high,4,4,med,high,vgood']
['low,high,4,4,big,low,unacc']
['low,high,4,4,big,med,acc']
['low,high,4,4,big,high,vgood']
['low,high,4,more,small,low,unacc']
['low,high,4,more,small,med,acc']
['low,high,4,more,small,high,acc']
['low,high,4,more,med,low,unacc']
['low,high,4,more,med,med,acc']
['low,high,4,more,med,high,vgood']
['low,high,4,more,big,low,unacc']
['low,high,4,more,big,me

In [11]:
import csv
with open("car.data",'r') as f:
    car_data=csv.reader(f,delimiter=',') # delimiter check all the time
    for i in car_data:
          curr.execute(f'insert into cardata.car values{tuple(i)}')
mydb.commit()

In [15]:
curr.execute('select * from cardata.car LIMIT 10')
curr.fetchall()

[('vhigh', 'vhigh', '2', '2', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '2', '2', 'big', 'high', 'unacc'),
 ('vhigh', 'vhigh', '2', '4', 'small', 'low', 'unacc')]

5. Try to group all the data with col1 and count occurance of each and every record based on col1 value?

In [13]:
curr.execute('SELECT buying, count(buying) from cardata.car group by buying')
curr.fetchall()

[('vhigh', 432), ('high', 432), ('med', 432), ('low', 432)]

6. Try to update a record where col3 value with 8 whereever you have value equal to 2?

In [28]:
Q = 'UPDATE cardata.car SET doors = "8" where doors = "2"'
curr.execute(Q)
mydb.commit()
curr.execute('select * from cardata.car where doors = "8" LIMIT 10')
curr.fetchall()

[('vhigh', 'vhigh', '8', '2', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '8', '2', 'big', 'high', 'unacc'),
 ('vhigh', 'vhigh', '8', '4', 'small', 'low', 'unacc')]

7. Try to filter where col3 value will be 4?

In [29]:
Q = 'select * from cardata.car where doors = "4" LIMIT 10'
curr.execute(Q)
curr.fetchall()

[('vhigh', 'vhigh', '4', '2', 'small', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'small', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'small', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'med', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'med', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'med', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'big', 'low', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'big', 'med', 'unacc'),
 ('vhigh', 'vhigh', '4', '2', 'big', 'high', 'unacc'),
 ('vhigh', 'vhigh', '4', '4', 'small', 'low', 'unacc')]

8. Try to dalete table?

In [31]:
Q = 'drop table cardata.car'
curr.execute(Q)
mydb.commit()
q = 'select * from cardata.car'
curr.fetchall()

ProgrammingError: 1051 (42S02): Unknown table 'cardata.car'

## Above error because car table has been deleted

9. Try to delete database?

In [32]:
Q = 'drop database cardata'
curr.execute(Q)
mydb.commit()
q = 'select * from cardata.car'
curr.fetchall()

[]

checked in sql worlbench database has been deleted